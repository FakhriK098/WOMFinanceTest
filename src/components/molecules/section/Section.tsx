import React from 'react';
import { StyleSheet, View } from 'react-native';
import Text from '@components/atoms/text/Text';
import { typography } from '@theme/typography';
import { colors } from '@theme/colors';
import { spacing } from '@theme/spacing';

type Props = {
  title: string;
  isDark: boolean;
  children: React.ReactNode;
};

const Section = ({ title, isDark, children }: Props) => {
  return (
    <View style={styles.container}>
      <Text
        weight="bold"
        size={typography.lg}
        color={isDark ? colors.text.dark : colors.text.light}
      >
        {title}
      </Text>
      <View style={styles.gap} />
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  container: { marginBottom: spacing.xl },
  gap: { height: spacing.sm },
});

export default React.memo(Section);
