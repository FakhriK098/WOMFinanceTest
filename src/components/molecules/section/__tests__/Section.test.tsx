import React from 'react';
import renderer, { act } from 'react-test-renderer';

import Section from '../Section';

describe('Section molecule', () => {
  it('matches snapshot and renders children', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(
        <Section title="Stats" isDark={false}>
          <></>
        </Section>,
      );
    });
    expect(tr!.toJSON()).toMatchSnapshot();
    const title = tr!.root.find((inst) => inst.props && inst.props.children === 'Stats');
    expect(title).toBeTruthy();
    act(() => tr!.unmount());
  });
});

