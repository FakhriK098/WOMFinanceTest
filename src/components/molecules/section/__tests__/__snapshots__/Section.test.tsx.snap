// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section molecule matches snapshot and renders children 1`] = `
<View
  style={
    {
      "marginBottom": 24,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#111827",
          "fontSize": 18,
          "fontWeight": "700",
        },
        undefined,
      ]
    }
  >
    Stats
  </Text>
  <View
    style={
      {
        "height": 8,
      }
    }
  />
</View>
`;
